### DISCUSSION TITLE
Noticing that version 3 is being written, what is its goal?

### DISCUSSION THREAD

**Discussion by litingyes** [koajs/koa#1749] [Q&A] [answered]
Noticing that version 3 is being written, what is its goal?

**Comment by zautumnz** [comment, marked as answer]
That's being tracked [here](https://github.com/koajs/koa/issues/1114). The main things, since v2 came out, were eventually removing any sort of support for v1 (generator-based) middleware, making any fixes that were _breaking_ changes, dropping support for truly ancient Node versions,	and _possibly_ moving to ESM (I think that's still being debated though?).
